src/python/pants/backend/helm/util_rules/chart.py

```diff
@@ -253,13 +260,32 @@ async def find_chart_for_deployment(request: FindHelmDeploymentChart) -> HelmCha
         ),
     )
 
-    found_charts = [tgt for tgt in explicit_targets if HelmChartFieldSet.is_applicable(tgt)]
+    first_party_charts = await MultiGet(
+        Get(HelmChart, HelmChartRequest, HelmChartRequest.from_target(tgt))
+        for tgt in explicit_targets
+        if HelmChartFieldSet.is_applicable(tgt)
+    )
+
+    third_party_charts: Iterable[HelmChart] = ()
+    fetch_third_party_request = FetchHelmArfifactsRequest.for_targets(
+        explicit_targets, description_of_origin=request.field_set.address.spec
+    )
+    if len(fetch_third_party_request.field_sets) > 0:
+        fetched_third_party_artifacts = await Get(
+            FetchedHelmArtifacts, FetchHelmArfifactsRequest, fetch_third_party_request
+        )
+        third_party_charts = await MultiGet(
```

Comment:
You could do both the first_party and third_party Gets in a single MultiGet. Something like:

```
    fetch_third_party_request = FetchHelmArfifactsRequest.for_targets(
        explicit_targets, description_of_origin=request.field_set.address.spec
    )
    fetched_third_party_artifacts = FetchedHelmArtifacts([])
    if fetch_third_party_request.field_sets:
        fetched_third_party_artifacts = await Get(
            FetchedHelmArtifacts, FetchHelmArfifactsRequest, fetch_third_party_request)

    charts = await MultiGet(
        *(Get(HelmChart, HelmChartRequest, HelmChartRequest.from_target(tgt))
        for tgt in explicit_targets
        if HelmChartFieldSet.is_applicable(tgt)),
        *(Get(HelmChart, FetchedHelmArtifact, fetched_artifact)
            for fetched_artifact in fetched_third_party_artifacts)
    )
```